addon/components/ui-component.js

```diff
@@ -43,3 +43,20 @@ export default Ember.Component.extend({
     );
   })
 });
+
+UIComponent.reopenClass({
+  uiStates: Ember.computed(function() {
+    let map = {};
+
+    this.eachComputedProperty((name, meta) => {
+      if (meta.isUIState) {
+        meta.name = name;
+        map[name] = meta;
+      }
+    });
+
+    return map;
+  }).readOnly()
```

Comment:
@ianstarz I don't think so. They are defined in the behavioral component.

For example:

```js
// addon/components/ui-button.js

export default UIComponent.extend({
  active: state(),
  browserActive: state(),
  disabled: state(),
  focus: state(),
  loading: state()  
})
```